app/services/solidus_bolt/transaction/refund.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module SolidusBolt
+  module Transaction
+    class Refund < SolidusBolt::Base
+      def initialize(transaction_reference:, amount:, currency:)
+        @transaction_reference = transaction_reference
+        @amount = amount
+        @currency = currency
+        super
+      end
+
+      def call
+        capture
+      end
+
+      private
+
+      def capture
+        options = build_options
+        HTTParty.post(
+          "#{api_base_url}/#{api_version}/merchant/transactions/credit",
+          options
+        )
+      end
+
+      def build_options
+        options = {
+          query: {
+            transaction_reference: @transaction_reference,
+            amount: @amount,
+            currency: @currency
+          },
+          headers: {
+            'X-Nonce' => generate_nonce
+          }
+        }
+
+        authenticate(options)
+        options
+      end
+
+      def generate_nonce
+        SecureRandom.hex
+      end
+
+      def authenticate(options)
+        options[:headers].merge!(authentication_header)
+      end
```

Comment:
snce these methods are repeated, what about moving them to the Base object?

Reply:
moved to base